src/compile/scale/domain.ts

```diff
@@ -249,6 +251,11 @@ function parseSingleChannelDomain(
   const {type} = fieldOrDatumDef;
   const timeUnit = fieldOrDatumDef['timeUnit'];
 
+  const dataSourceTypeForScaleDomain = getScaleDataSourceForHandlingInvalidValues({
```

Comment:
FYI, this is the main change in this file. Instead of always drawing scale domains from "main" data source, we may choose pre-/post- filtering invalid values, depending on the invalid data mode.